front/app/modules/commercial/insights/admin/containers/Insights/List/CreateInsightsView.tsx

```diff
@@ -99,10 +179,75 @@ export const CreateInsightsView = ({
     }
   };
 
+  const toggleSelectProject = (projectId: string) => {
+    if (selectedProjectsIds.includes(projectId)) {
+      setSelectedProjectsIds(
+        selectedProjectsIds.filter((id) => id !== projectId)
+      );
+    } else {
+      setSelectedProjectsIds([...selectedProjectsIds, projectId]);
+    }
+  };
+  const toggleExpandFolder = (folderId: string) => {
+    if (expandedFoldersIds.includes(folderId)) {
+      setExpandedFoldersIds(expandedFoldersIds.filter((id) => id !== folderId));
+    } else {
+      setExpandedFoldersIds([...expandedFoldersIds, folderId]);
+    }
+  };
+
+  // Transform folders data to include projects
+  const folders = !isNilOrError(projectFolders)
```

Comment:
Might be premature optimization, but this seems like a perfect case for `useMemo`

Reply:
Also maybe a more descriptive variable name would be `foldersIncludingProjects`

Reply:
I do think it makes sense so I added useMemo but the effectiveness of the optimization depends also on how well the dependencies are memoized which I didn't go too far into investigating.